client/state/jetpack-connection-health/selectors/is-jetpack-connection-unhealthy.js

```diff
@@ -4,19 +4,19 @@ import 'calypso/state/jetpack-connection-health/init';
  * Returns true if the current site Jetpack connection was checked and is unhealthy
  * @param  {Object}  state         Global state tree
  * @param  {number}  siteId        Site ID
- * @returns {?boolean}             Whether the current site can have connection problem
+ * @returns {boolean}             Whether the current site can have connection problem
  */
 export default function isJetpackConnectionUnhealthy( state, siteId ) {
-	if ( ! siteId ) {
-		return null;
+	const siteState = state.jetpackConnectionHealth[ siteId ];
+
+	if ( ! siteState?.connectionHealth ) {
+		return false;
 	}
-	const connection_data = state.jetpackConnectionHealth?.[ siteId ];
 
-	if ( ! connection_data ) {
-		return null;
+	let error = siteState.connectionHealth.error;
+	if ( siteState.connectionHealth.error === undefined ) {
+		error = '';
 	}
 
-	return (
-		true === connection_data.jetpack_connection_problem && false === connection_data.is_healthy
-	);
+	return true === siteState.connectionHealth.jetpack_connection_problem && '' !== error;
```

Comment:
This empty string check is a bit messed up. Would have been much better if we deal with `null` or `undefined` in that case.